src/Frontend/MainWindow.vala

```diff
@@ -221,25 +234,30 @@ class Taxi.MainWindow : Hdy.ApplicationWindow {
     }
 
     private void update_bookmark_menu () {
-        foreach (Gtk.Widget child in bookmark_list.get_children ()) {
-            child.destroy ();
+        for (Gtk.Widget? child = bookmark_list.get_first_child (); child != null;) {
+            Gtk.Widget? next = child.get_next_sibling ();
+            bookmark_list.remove (child);
+            child = next;
         }
 
         var uri_list = conn_saver.get_saved_conns ();
         if (uri_list.length () == 0) {
             bookmark_menu_button.sensitive = false;
         } else {
             foreach (string uri in uri_list) {
-                var bookmark_item = new Gtk.ModelButton ();
-                bookmark_item.text = uri;
-
-                bookmark_list.add (bookmark_item);
-
+                var bookmark_item = new Gtk.Button () {
+                    child = new Gtk.Label (uri) {
+                        halign = START
+                    }
+                };
+                bookmark_item.add_css_class (Granite.STYLE_CLASS_MENUITEM);
                 bookmark_item.clicked.connect (() => {
                     connect_box.go_to_uri (uri);
+                    bookmark_popover.popdown ();
```

Comment:
Doesn't `Gtk.Popover.autohide` work for you instead of changing the scope of `bookmark_popover` from a local variable to a private member variable and manually popdown here?